### DISCUSSION TITLE
Grafana empty data after upgrade

### DISCUSSION THREAD

**Discussion by arjoshi89** [Security-Onion-Solutions/securityonion#6571] [Unsupported Versions] [answered]
Issue:
- grid status showing fault under UI
-Grafana empty data after upgrade

1 master node, 2 heavy nodes
upgraded from 2.3.10 to 2.3.90

*********************************************
- Run below commands under master nodes:

salt \* test.ping
securityonion_manager:
    True
sensor2_heavynode:
    True
sensor1_heavynode:
    True

Checking Docker status

    Docker --------------------------------------------------------------------------------------------------------------------------------------- [ OK ]

Checking container statuses

    so-aptcacherng ------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-cortex ------------------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-dockerregistry ---------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-elastalert -------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-elasticsearch ----------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-filebeat ---------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-fleet ------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-grafana ----------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-idstools ---------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-influxdb ----------------------------------------------------------------------------------------------------------------------------- [ MISSING ]
    so-kibana ------------------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-kratos ------------------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-logstash ---------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-mysql ------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-nginx ------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-playbook ---------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-redis ------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-sensoroni --------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-soc --------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-soctopus ---------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-telegraf ----------------------------------------------------------------------------------------------------------------------------- [ MISSING ]
    so-thehive ----------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-thehive-es -------------------------------------------------------------------------------------------------------------------------------- [ OK ]

so-status showing influxdb and telegraf missing. Influxdb failed on restart

influxdb logs on master:

ts=2021-12-15T02:43:33.206781Z lvl=info msg="Registered diagnostics client" log_id=0YQ~PnsG000 service=monitor name=network
ts=2021-12-15T02:43:33.206808Z lvl=info msg="Registered diagnostics client" log_id=0YQ~PnsG000 service=monitor name=system
ts=2021-12-15T02:43:33.206856Z lvl=info msg="Starting precreation service" log_id=0YQ~PnsG000 service=shard-precreation check_interval=10m advance_period=30m
ts=2021-12-15T02:43:33.206893Z lvl=info msg="Starting snapshot service" log_id=0YQ~PnsG000 service=snapshot
ts=2021-12-15T02:43:33.206915Z lvl=info msg="Starting continuous query service" log_id=0YQ~PnsG000 service=continuous_querier
ts=2021-12-15T02:43:33.206937Z lvl=info msg="Starting HTTP service" log_id=0YQ~PnsG000 service=httpd authentication=false
ts=2021-12-15T02:43:33.206956Z lvl=info msg="Storing statistics" log_id=0YQ~PnsG000 service=monitor db_instance=_internal db_rp=monitor interval=10s
run: open server: open service: tls: private key does not match public key

***********************

so-status showing OK under heavy nodes

Checking Docker status

    Docker --------------------------------------------------------------------------------------------------------------------------------------- [ OK ]

Checking container statuses

    so-curator ----------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-elasticsearch ----------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-filebeat ---------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-logstash ---------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-nginx ------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-redis ------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-sensoroni --------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-steno ------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-strelka-backend --------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-strelka-coordinator ----------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-strelka-filestream ------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-strelka-frontend -------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-strelka-gatekeeper ------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-strelka-manager --------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-suricata ---------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-telegraf ---------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-zeek -------------------------------------------------------------------------------------------------------------------------------------- [ OK ]

**************************************

**Comment by m0duspwnens** [comment]
@arjoshi89 Can you post your `soup.log`?

**Comment by arjoshi89** [comment]
Attached file-soup.log

**Comment by m0duspwnens** [comment]
On your manager, can you run the following and verify if the md5 matches?
`openssl rsa -noout -modulus -in /etc/pki/influxdb.key | openssl md5`
`openssl x509 -noout -modulus -in /etc/pki/influxdb.crt | openssl md5`

**Reply by arjoshi89** [reply]
Yes, it matches exactly

**Reply by m0duspwnens** [reply]
I am in the process of trying to replicate this in the lab, but in the meantime, can you check the following?

On the manager run:
`openssl x509 -in /etc/pki/influxdb.crt -text -noout`

Look for the lines `Issuer:` and `Subject:` and verify the CN matches the hostname of your manager.

Look for the line `X509v3 Subject Alternative Name:` and under, verify `DNS:` matches the manager hostname. Also ensure that `IP Address:` matches IP of the manager.

Also, from the manager, run:
`salt-call pillar.get global:managerip` and ensure it matches the IP of the manager.

If you do a hostname lookup for the manager, from the manager, do you get the proper IP?

Thanks.

**Comment by arjoshi89** [comment, marked as answer]
Certificate date shows updated start date as upgrade date.
Openssl command-> CN and DNS matches manager hostname. 
salt command also gives correct manager IP

Thank you for looking in lab

**Reply by m0duspwnens** [reply]
I was unable to replicate this in the lab. One last thing I would try forcing the `/etc/pki/influxdb.key` and `/etc/pki/influxdb.crt` to be recreated.

On the manager:

-  Ensure that `so-telegraf` and `so-influxdb` are not running.
-  Move the `influxdb.key` and `influxdb.crt` to a safe location for backup.
-  Apply the ssl state with `salt-call state.apply ssl -linfo`, you should see `influxdb.crt` and `influxdb.key` get created during the state run.
-  Apply the `influxdb` state with `salt-call state.apply influxdb -linfo`. This should start the `so-influxdb` container, then test the connection after. The test should fail for a bit while waiting for InfluxDB to fully to start, then eventually succeed.
-  Apply the `telegraf` state with `salt-call state.apply telegraf -linfo`. This should start `so-telegraf` container.
-  Check the output of `so-status`.
-  If they are still failing, check the logs at `/opt/so/log/influxdb/influxdb.log` and `/opt/so/log/telegraf/telegraf.log`.

**Reply by arjoshi89** [reply]
Applied the command as suggested. It fixed the issue.
Thank you for your help !
